From 611b66bde5154057c4ce22e4ed1956f221d9e687 Mon Sep 17 00:00:00 2001
From: Guillaume <guillaume.thouvenin@vates.tech>
Date: Mon, 31 Mar 2025 13:01:54 +0200
Subject: [PATCH] [xcp-ng] add debug info in observer

This patch adds traces to see observer not working on SMAPI component.

Signed-off-by: Guillaume <guillaume.thouvenin@vates.tech>
---
 ocaml/xapi/xapi_observer_components.ml | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/ocaml/xapi/xapi_observer_components.ml b/ocaml/xapi/xapi_observer_components.ml
index 0b3b884f4..e78031891 100644
--- a/ocaml/xapi/xapi_observer_components.ml
+++ b/ocaml/xapi/xapi_observer_components.ml
@@ -90,9 +90,17 @@ let is_component_enabled ~component =
                  |> List.mem component
             )
             observers
-        with _ -> false
+        with e ->
+          D.log_backtrace () ;
+          D.warn "is_component_enabled(%s) inner got exception: %s"
+            (to_string component) (Printexc.to_string e) ;
+          false
       )
-  with _ -> false
+  with e ->
+    D.log_backtrace () ;
+    D.warn "is_component_enabled(%s) got exception: %s" (to_string component)
+      (Printexc.to_string e) ;
+    false
 
 let is_smapi_enabled () = is_component_enabled ~component:SMApi
 
